test/e2e/network/dual_stack.go

```diff
@@ -240,11 +246,16 @@ var _ = SIGDescribe("[Feature:IPv6DualStackAlphaFeature] [LinuxOnly]", func() {
 		validateServiceAndClusterIPFamily(svc, defaultIPFamily)
 
 		// ensure endpoint belong to same ipfamily as service
-		endpoint, err := cs.CoreV1().Endpoints(svc.Namespace).Get(svc.Name, metav1.GetOptions{})
-		if err != nil {
+		if err := wait.PollImmediate(500*time.Millisecond, 10*time.Second, func() (bool, error) {
+			endpoint, err := cs.CoreV1().Endpoints(svc.Namespace).Get(svc.Name, metav1.GetOptions{})
+			if err != nil {
+				return false, nil
```

Comment:
Shouldn't we return the error here is it's not nil?

Reply:
if you return an error it stops polling, you just returns false to retry

https://github.com/kubernetes/kubernetes/blob/dd77283426bd3249004472efcbb3257744448161/staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go#L323-L332

Reply:
ok, sounds good! In that case can we add a log here to print the error?

Reply:
done